coverage/artifacts/local.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package artifacts
+
+import (
+	"os"
+	"path"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	"io"
+	io2 "k8s.io/test-infra/coverage/io"
+	"k8s.io/test-infra/coverage/logUtil"
+)
+
+//LocalArtifacts sub-type of Artifacts. Represent artifacts stored locally (
+// as oppose to artifacts stored in GCS bucket)
+type LocalArtifacts struct {
```

Comment:
type LocalArtifacts Artifacts does not work the same way. It will not keep any methods from Artifacts; yet those methods (e.g. .ProfilePath) are needed.

Reply:
That is true but this still seems backwards. Normally you'd do something like have an `type Artifacts interface` and have the different variants meet the interface, then compose generic code on top of the interface.

Reply:
In this case, methods of LocalArtifacts has to access data stored in the underlying Artifacts struct. Interface cannot do that

Reply:
@BenTheElder